packages/back-end/src/controllers/experiments.ts

```diff
@@ -2217,6 +2273,23 @@ export async function postVisualChangeset(
     throw new Error("Could not find experiment");
   }
 
+  if (req.body.urlRedirects) {
+    const variationIds = experiment?.variations.map((v) => v.id);
+    const reqVariationIds = req.body.urlRedirects.map((r) => r.variation);
+
+    const areValidVariations = variationIds?.every((v) =>
+      reqVariationIds.includes(v)
+    );
+    if (!areValidVariations) {
+      throw new Error("Invalid variation IDs for urlRedirects");
+    }
+
+    const origin = req.body.urlPatterns;
+    const destinations = req.body.urlRedirects;
+
+    await _validateRedirect(origin, destinations, context, experiment);
```

Comment:
Could consider using a param after this first errors if the user decides to continue anyway